omod/src/main/java/org/openmrs/web/controller/ForgotPasswordFormController.java

```diff
@@ -123,12 +121,20 @@ protected ModelAndView onSubmit(HttpServletRequest request, HttpServletResponse
 				finally {
 					Context.removeProxyPrivilege(PrivilegeConstants.GET_USERS);
 				}
-				
-				if (user == null /*|| StringUtils.isEmpty(user.getSecretQuestion())*/) {
+
+				String secretQuestion = null;
+				if(user != null) {
+					secretQuestion = Context.getUserService().getSecretQuestion(user);
+				}
+
+				if (user == null) {
```

Comment:
I think you merge these 2 if clauses because you're checking if user is not null above and here you're checking if they are null